app/services/pagelets/manager.rb

```diff
@@ -0,0 +1,35 @@
+module Pagelets
+  class Manager
+    class << self
+
+      def add_pagelet(page_name, mountpoint, opts)
+        handle_empty_keys_for page_name, mountpoint
+        raise ::Foreman::Exception.new("Cannot add pagelet to page #{page_name} without partial") unless opts[:partial]
+        raise ::Foreman::Exception.new("Cannot add pagelet to page #{page_name} without mountpoint") if mountpoint.nil?
+        priority = opts[:priority] || default_pagelet_priority(page_name, mountpoint) - 100
```

Comment:
I guess so, this feels more natural to me: 1, 2 3 :-)

One more comment for this block - can you also store whole `options` hash in Pagelet itself? I plan to pass some variables to my plugin from the initializer block and I want to use it on the ERB page.